bundles/framework/divmanazer/component/Chart.js

```diff
@@ -49,58 +59,53 @@ Oskari.clazz.define('Oskari.userinterface.component.Chart', function(sandbox, lo
         return dimensions;
     },
     createGridlines: function () {
-        return d3.axisBottom(this.x).ticks(10);
+        return d3.axisBottom( this.x ).ticks(10);
     },
     initScales: function () {
         this.x = d3.scaleLinear()
-        .domain([ d3.min( this.data, function (d) {
+        .domain([ d3.min( this.data, function ( d ) {
                 return d.value;
-            }), d3.max( this.data, function (d) {
+            }), d3.max( this.data, function ( d ) {
                 return d.value;
             })
         ]);
 
         this.y = d3.scaleBand()
-        .domain(this.data.map(function (d) {
+        .domain( this.data.map ( function ( d ) {
             return d.name;
         }));
     },
-    initSelection: function() {
-            var svg = d3.select(this._template.graph[0]).append("svg")
-                .attr("width", this.dimensions.width() + this.dimensions.margin.left + this.dimensions.margin.right)
-                .attr("height", this.dimensions.height() + this.dimensions.margin.top + this.dimensions.margin.bottom)
-                .append("g")
-                .attr("transform", "translate(" + this.dimensions.margin.left + "," + this.dimensions.margin.top + ")"
-            );
-            this.svg = svg;
+    initSVG: function () {
+        var svg = d3.select( this._template.graph[0] ).append("svg")
```

Comment:
Use this._template.graph.clone() instead. That way you can remove any this._template.graph.empty() calls.

Reply:
d3.select needs a string or a dom-node to append to, so jQuery won't work here.
https://github.com/d3/d3-selection/blob/master/README.md#select

Reply:
But if you clone() the DOM-element generated before passing it to d3 you stop modifying the template variable and won’t need to ”clear()” it in other pieces of the code. So either clone it or stop calling it template.